### ISSUE TITLE
There is no sound for push notifications.

### ISSUE THREAD

**Issue by YamatoRyou** [deltachat/deltachat-desktop#2449]
- Operating System: x64 Windows 7
- Delta Chat Version: 1.25.1 (git: v1.25.0-2-g070ab539)
- Expected behavior: When the notification appears in the lower right corner of the desktop, a beep should sound. The style of the beep depends on the system settings. Just like Windows 10.
- Actual behavior: Under Windows 7, push notifications do not have any sound for prompting.
- Steps to reproduce the problem: n/a
- Screenshots: n/a
- Logs: n/a

@gerryfrancis

**Comment by gerryfrancis** [comment]
Thank you for your report. :+1: Unfortunately I have no Windows 7 system available at the moment to verify. Can you hear the notification sound when using a different mail app than Delta Chat?

**Comment by YamatoRyou** [comment]
> Thank you for your report. +1 Unfortunately I have no Windows 7 system available at the moment to verify. Can you hear the notification sound when using a different mail app than Delta Chat?

Yes, I have used the following two mail clients, they can sound normally when they receive new mail under Windows 7. And there are different strategies for the choice of sound:
Foxmail: Use the sound from the operating system settings (the sound style is part of the theme);
![{72715A1F-C5A4-7DC4-F015-986A51389D53}](https://user-images.githubusercontent.com/48104960/144327151-e0da86df-0755-41b9-ae0b-23446fa1e9ed.png)
![{184A5F92-53F6-8AC2-EFDE-E0DA984E8838}](https://user-images.githubusercontent.com/48104960/144326975-e340a4ed-47d2-4e49-a7ec-9e18aa411f47.png)

DreamMail Pro: Use software customized voices.
![{E33F16FD-7531-8232-A66D-A19D4FF6B979}](https://user-images.githubusercontent.com/48104960/144327394-57375366-461a-4289-b7fa-d1af4306fc38.png)

It seems that the notification style of Delta Chat is self-implemented under Windows 7, and an independent sound mechanism can be used.

**Comment by gerryfrancis** [comment]
@YamatoRyou Thank you for your input. The sound is not played in Windows 8.1 (which I use right now) either. In fact, I do not get a visible notification for incoming messages at all, can you confirm, please?

**Comment by gerryfrancis** [comment]
Seems like Delta Chat is missing notification support in Windows 8.1 at all, at least I cannot tick the app in Windows settings for notifications:

![grafik](https://user-images.githubusercontent.com/51095533/144332533-19d652e4-d5b2-4bd7-8098-8687eef8bcd2.png)

**Comment by YamatoRyou** [comment]
> 似乎 Delta Chat 完全缺少 Windows 8.1 中的通知支持，至少我无法在 Windows 设置中勾选应用程序的通知：
> 
> ![涂鸦](https://user-images.githubusercontent.com/51095533/144332533-19d652e4-d5b2-4bd7-8098-8687eef8bcd2.png)

Same.
![{666F4464-D7F6-661E-8C71-68006C7B8823}](https://user-images.githubusercontent.com/48104960/144334337-ebbd9433-d7d4-42ad-b9e4-d08adbfa80e9.png)

**Comment by gerryfrancis** [comment]
@YamatoRyou Thanks for your confirmation, but ... are you using Windows 7 or Windows 8.1 with Delta Chat?

**Comment by YamatoRyou** [comment]
> @YamatoRyou Thanks for your confirmation, but ... are you using Windows 7 or Windows 8.1 with Delta Chat?

Windows 7 is a system I often use, and Windows 8.1 is temporarily installed on VMWare to assist you in testing.

**Comment by gerryfrancis** [comment]
@YamatoRyou Ah, OK! :) Back to your issue in Windows 7 then: Do you get visual notifications (while the Delta Chat window is not focused)?

**Comment by YamatoRyou** [comment]
> @YamatoRyou Ah, OK! :) Back to your issue in Windows 7 then: Do you get visual notifications of Delta Chat (while the app window is not focused)?

Sure.
![{80347BEB-7C18-88DC-A8DC-A39C7A720450}](https://user-images.githubusercontent.com/48104960/144337959-91df7018-cd72-489f-a0b3-164b9ce5654e.png)

**Comment by gerryfrancis** [comment]
@YamatoRyou Interesting, this would mean that Delta Chat lacks support only for notification sounds on Windows 7, but does not support Windows 8.1 notifications at all.

Maybe an old Windows notification library included in the Delta Chat project?

**Comment by YamatoRyou** [comment]
> @YamatoRyou Interesting, this would mean that Delta Chat lacks support only for notification sounds on Windows 7, but does not support Windows 8.1 notifications at all.
> 
> Maybe an old Windows notification library included in the Delta Chat project?

The optimal solution for this problem can refer to the Telegram for Windows desktop version.
All supported Windows versions will use their own notification push system, but also freely adjust the style of the notification; sound; screen position; frequency and repeated reminder.

**Comment by Jikstra** [comment]
I looked into it a bit with @tommi3141 and it seems there is a node module for doing notifications on windows. Probably this helps with some of the issues we have?
https://www.google.com/search?channel=trow5&client=firefox-b-d&q=electron+windows+notifications

**Comment by gerryfrancis** [comment]
@Jikstra It is not released for Windows 7, only Windows 8 and above, but it looks nice! :+1: Do you have an idea why the sound is not played in 7?

**Comment by Jikstra** [comment]
> @Jikstra It is not released for Windows 7, only Windows 8 and above, but it looks nice! +1 Do you have an idea why the sound is not played in 7?

Oh. Good hint ^^ Ehm I don't know exactly, but probably the chrome browser or electron devs never gave a shit about implementing web notifications api properly for windows 7? Win 7 is end of life since almost a year.

**Comment by gerryfrancis** [comment]
> Win 7 is end of life since almost a year.

In January 2022 EOL for Windows 7 will be for two years. Although we support 7 and up, maybe even Windows 11 inofficially(?), should we stick to supporting 7 any longer? Even 8.1 will have its EOL next month, FYI.

**Comment by Jikstra** [comment]
> > Win 7 is end of life since almost a year.
> 
> In January 2022 EOL for Windows 7 will be for two years. Although we support 7 and up, maybe even Windows 11 inofficially(?), should we stick to supporting 7 any longer? Even 8.1 will have its EOL next month, FYI.

I think there can be some point of supporting old OSs/architectures (32bit). There is old hardware around, and continuing to use it is the most sustainable act from a recycling perspective. Throwing away a computer because win11 doesn't run on it and buying a new one is a problem on many levels (money, capitalism, climate change..., respect of resources/work). But of course supporting old things is not without effort, someone has to maintain/implement it and I think this is where things are getting interesting. If somebody wants to add support for this, I'm happy to merge it. Maybe I will even do it myself because this is important to me too, but I mainly focus on linux and not so much on windows. Personal thing, but I'm open for the bigger perspective of supporting as much as we are able to maintain. And If we're not able to maintain this anymore, we have to drop it or keep it at the (semi) broken state.

**Comment by Simon-Laux** [comment]
I think we just need to play a sound ourselves on platforms where there isn't one already (also have a toggle in the settings for the case that our assumptions are wrong so that the user can disable it).

Also I highly suspect that we need to probably set some registry keys while installing to get DC to appear in those notification settings or use an appx/UWP format/api for it.
We need to look into the registry keys anyway so people can select Deltachat as default mailto handler in the settings.

**Comment by Simon-Laux** [comment]
in windows 10 the sounds works for me using current master (1.28.1, plaese wait for 1.28.2 when updating, currently custom chat background is broken on windows)

**Comment by gerryfrancis** [comment]
@Simon-Laux Notification centers in Windows 7, 8(.1), and 10 are quite different to each other. E.g. I tested DC version 1.28.1, and I receive no visual notification for new messages at all.

Values for new mail notification sounds can be found in the subkeys of `HKEY_CURRENT_USER\AppEvents\Schemes\Apps\.Default\MailBeep` .

**Comment by Simon-Laux** [comment]
Is also to be a problem on linux: https://github.com/deltachat/deltachat-desktop/issues/3289#issuecomment-1606305485

**Comment by Simon-Laux** [comment]
Let's reopen this if it still is an issue on windows 11. (windows 10 is EOL in October)
